### PR TITLE
Add filters to maestro_subscription_health

### PR DESCRIPTION
**Pull request by lewing**
## Summary
- add staleOnly, channelFilter, sourceRepoFilter, and compact to maestro_subscription_health
- keep filters after the parallel health fan-out and preserve no-arg detailed output
- extract filter/formatter helpers and add direct tests
- update Naomi history, decision inbox, and mcp-tool-filtering skill confidence/examples

## Validation
- dotnet test --no-restore --nologo (188 passed; one earlier cache expiry timing failure passed on rerun)
- live dotnet/dotnet measurement: detailed 24,398 bytes; staleOnly+compact 3,049 bytes for 93 subscriptions / 43 stale

Stacked on PR #22 because that channel-filter branch is still the current base for this pattern.

### CODE REVIEW

File: src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs [hunk 1]
```diff
@@ -97,25 +97,67 @@ public async Task<string> GetSubscription(
...
+    internal static IEnumerable<SubscriptionHealthResult> FilterSubscriptionHealthResults(
+        IEnumerable<SubscriptionHealthResult> results,
+        bool staleOnly = false,
+        string? channelFilter = null,
+        string? sourceRepoFilter = null)
+    {
+        var filtered = results;
+
+        if (staleOnly)
+            filtered = filtered.Where(r => r.IsStale);
```
[the remark this answers is not in the document]
Comment (lewing) [reply src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs]: Fixed in edabfea: staleOnly now treats Error != null as non-healthy and includes those rows, with a unit test covering errored subscriptions.

File: src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs [hunk 2]
```diff
@@ -214,7 +256,59 @@ public async Task<string> GetSubscriptionHealth(
...
+    {
+        var source = ShortRepositoryName(result.SourceRepository);
+        var status = result.IsStale
+            ? FormatCompactStaleStatus(result)
+            : "current";
+        var pr = CompactPrReference(result.TrackedPr?.PrUrl);
+        var prefix = result.Error != null ? "❌" : result.IsStale ? "⚠️" : "✅";
+        var error = result.Error != null ? $"; error: {result.Error}" : "";
+
+        return $"{prefix} {source} → {result.TargetBranch}: {status} (PR: {pr}{error})";
```
[the remark this answers is not in the document]
Comment (lewing) [reply src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs:274]: Fixed in edabfea: compact output now renders errored rows with status "error" instead of "current", with a regression test.

File: src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs [hunk 3]
```diff
@@ -214,7 +256,59 @@ public async Task<string> GetSubscriptionHealth(
...
+            ? trimmed[..^4]
+            : trimmed;
+
+        var parts = trimmed.Split('/', StringSplitOptions.RemoveEmptyEntries);
+        return parts.Length switch
+        {
+            >= 2 => string.Join('/', parts[^2..]),
+            1 => parts[0],
+            _ => repository
+        };
```
[the remark this answers is not in the document]
Comment (lewing) [reply src/MaestroTool.Core/MaestroMcpTools/MaestroMcpTools.Subscriptions.cs]: Fixed in edabfea: ShortRepositoryName now special-cases AzDO URLs via MaestroService.ParseAzDoUrl and displays/filters them as org/project/repo, with GitHub/AzDO/edge-case tests.
